include/nbl/builtin/hlsl/bxdf/ndf/common.hlsl

```diff
@@ -56,6 +57,154 @@ float microfacet_to_light_measure_transform(in float NDFcos_already_in_reflectiv
+        {
+            return geom_smith::VNDF_pdf_wo_clamps(d, lambda_V.absNdotV, transmitted, VdotH, LdotH, VdotHLdotH, orientedEta, reflectance, onePlusLambda_V);
+        }
+        static scalar_t VNDF_static(in scalar_t d, in scalar_t G1_over_2NdotV)
+        {
+            return geom_smith::VNDF_pdf_wo_clamps(d, G1_over_2NdotV);
+        }
+
+        static scalar_t VNDF_fromLambda_impl(in scalar_t d, in scalar_t lambda, in float maxNdotV)
+        {
+            float dummy;
+            return VNDF_static(d, lambda, maxNdotV, dummy);
+        }
+        static scalar_t VNDF_fromG1_over_2NdotV_impl(in scalar_t d, in scalar_t G1_over_2NdotV)
+        {
+            return VNDF_static(d, G1_over_2NdotV);
+        }
+
+        
+
+        // VNDF isotropic variants
+        scalar_t VNDF(in float NdotH2, in float NdotV2, in float maxNdotV)
+        {
+            const float d = ndf.D(NdotH2);
+            const float lambda = ndf.Lambda(NdotV2);
+            return VNDF_fromLambda_impl(d, lambda, maxNdotV);
+        }
+        scalar_t VNDF(in float NdotH2, in float G1_over_2NdotV)
+        {
+            const float d = ndf.D(NdotH2);
+            return VNDF_fromG1_over_2NdotV_impl(d, G1_over_2NdotV);
+        }
+        scalar_t VNDF(
+            in float NdotH2, in float NdotV2, 
+            in float absNdotV, in bool transmitted, in float VdotH, in float LdotH, in float VdotHLdotH, in float orientedEta, in float reflectance, out float onePlusLambda_V)
+        {
+            const float d = ndf.D(NdotH2);
+            const float lambda = ndf.Lambda(NdotV2);
+
+            return VNDF_static(d, lambda, absNdotV, transmitted, VdotH, LdotH, VdotHLdotH, orientedEta, reflectance, onePlusLambda_V);
+        }
+
+        // VNDF anisotropic variants
+        scalar_t VNDF(
+            in float TdotH2, in float BdotH2, in float NdotH2,
+            in float TdotV2, in float BdotV2, in float NdotV2,
+            in float maxNdotV)
+        {
+            const float d = ndf.D(TdotH2, BdotH2, NdotH2);
+            const float lambda = ndf.Lambda(TdotV2, BdotV2, NdotV2);
+            return VNDF_fromLambda_impl(d, lambda, maxNdotV);
+        }
+        scalar_t VNDF(
+            in float TdotH2, in float BdotH2, in float NdotH2, 
+            in float G1_over_2NdotV)
+        {
+            const float d = ndf.D(TdotH2, BdotH2, NdotH2);
+            return VNDF_fromG1_over_2NdotV_impl(d, G1_over_2NdotV);
+        }
+        scalar_t VNDF(
+            in float TdotH2, in float BdotH2, in float NdotH2,
+            in float TdotV2, in float BdotV2, in float NdotV2,
+            in float absNdotV, in bool transmitted, in float VdotH, in float LdotH, in float VdotHLdotH, in float orientedEta, in float reflectance, out float onePlusLambda_V)
+        {
```

Comment:
if it makes sense, delcare parameter structs in the `bxdf` namespace for the ndf functions and gather some of these inputs together, right now its tedious to read and typo prone (many parameters have the same type, you'll die)